Sources/SourceControl/RepositoryManager.swift

```diff
@@ -306,6 +319,33 @@ public class RepositoryManager {
         }
     }
 
+    /// Fetches the repository into the cache. If no `cachePath` is set or an error ouccured fall back to fetching the repository without populating the cache.
+    /// - Parameters:
+    ///   - handle: The specifier of the repository to fetch.
+    ///   - update: Update a repository that is already cached or fetch the repository into the cache.
+    ///   - repositoryPath: The path where the repository should be fetched to.
+    func fetchAndPopulateCache(handle: RepositoryHandle, repositoryPath: AbsolutePath, update: Bool) throws {
+        if let cachePath = cachePath {
+            let cachedRepositoryPath = cachePath.appending(component: handle.repository.fileSystemIdentifier)
+            do {
+                try initalizeCacheIfNeeded(cachePath: cachePath)
+                try fileSystem.withLock(on: cachedRepositoryPath, type: .exclusive) {
+                    // Fetch the repository into the cache.
+                    try self.provider.fetch(repository: handle.repository, to: cachedRepositoryPath, update: update)
+                    // Copy the repository from the cache into the repository path.
+                    try self.provider.copy(from: cachedRepositoryPath, to: repositoryPath)
+                }
+            } catch {
+                // Fetch without populating the cache in the case of an error.
+                print("Skipping cache due to an error: \(error)")
```

Comment:
Probably need to find a way to surface this error in some other way than a print statement (for clients such as IDEs, etc).  Not sure yet if that means adding a delegate here or something like that.

Reply:
I think we can keep this as a print for now. Since this is an unimportant message, which does not affect the user directly e.g. the fetch will still work perfectly fine. It's just a hint for us. This will also appear in the build logs, right?

Reply:
I think this is fine to clear up in a future PR, in the interest of getting this integrated.  From `swift` CLI tools, it will be intermixed with the other `stdout` output... but for clients of libSwiftPM, print statements don't show up in any build logs.  Only content emitted to `DiagnosticsEngine`s do.  The output will instead appear on the `stdout` of whatever tool or app is linking against libSwiftPM.